apps/Standalone/src/designer/app/AzureLogicAppsDesigner/laDesigner.tsx

```diff
@@ -395,6 +398,29 @@ const DesignerEditor = () => {
           ...connectionsData?.agentConnections,
           ...newAgentConnections,
         };
+
+        // Assign MSI roles if needed
+        /**
+         *  This is currently only for Agentic workflows,
+         *    but we should work to make this generic in the future
+         *  The issue with making it generic is that we don't have a good way of getting the required definition names for any given connection reference
+         *  The required roles are listed on connection parameters which we don't have access to here,
+         *    and would take several requests to check for each connection, when most will not need it, leading to unnecessary slowdown during save
+         *  One option is to populate that info somewhere in the connection reference for use here,
+         *    but that is unavailable at authoring time when we are populating the values that require the roles
+         */
+        for (const [refKey, agentConnection] of Object.entries(newAgentConnections)) {
+          const reference = connectionReferences?.[refKey];
+          if (reference?.authentication?.type === 'ManagedServiceIdentity') {
+            const definitionNames = ['Azure AI Administrator', 'Cognitive Services Contributor'];
+            const missingRoleAssignments = await getMissingRoleDefinitions(agentConnection?.resourceId, definitionNames);
+            const assignmentPromises = [];
+            for (const roleDefinition of missingRoleAssignments) {
+              assignmentPromises.push(RoleService().addRoleAssignmentForApp(agentConnection?.resourceId, roleDefinition.id));
+            }
+            await Promise.all(assignmentPromises);
```

Comment:
NIT: I know this is standalone, but just in case you follow the same pattern in portal. Are we ok rejecting all of the promises if one of them fails? Should we log the error in telemetry? should we show an error if one of the api call fails? (Letting the user know that the role assignment failed but saving the workflow went well?